### PR TITLE
[Ready for Review] Adds Player Notes

### PR DESCRIPTION
**Pull request by GeneriedJenelle**
Ports from baystation, will update as this goes along.

### CODE REVIEW

File: code/modules/admin/player_notes.dm [hunk 1]
```diff
@@ -0,0 +1,83 @@
+//This stuff was originally intended to be integrated into the ban-system I was working on
+//but it's safe to say that'll never be finished. So I've merged it into the current player panel.
+//enjoy				~Carn
+
+#define NOTESFILE "data/player_notes.sav"	//where the player notes are saved
+
+obj/admins/proc/notes_show(var/ckey)
```
Comment (TorinoFermic) [inline code/modules/admin/player_notes.dm]: obj is missing the first / at line 7.

File: code/modules/admin/player_notes.dm [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	usr << browse("<head><title>Player Notes</title></head><body>[notes_gethtml(ckey)]</body>","window=player_notes;size=700x400")
+
+
+obj/admins/proc/notes_gethtml(var/ckey)
```
Comment (TorinoFermic) [inline code/modules/admin/player_notes.dm]: Also this too.

File: code/modules/admin/player_notes.dm [hunk 3]
```diff
@@ -0,0 +1,83 @@
+//This stuff was originally intended to be integrated into the ban-system I was working on
+//but it's safe to say that'll never be finished. So I've merged it into the current player panel.
+//enjoy				~Carn
+
+#define NOTESFILE "data/player_notes.sav"	//where the player notes are saved
+
+obj/admins/proc/notes_show(var/ckey)
+	usr << browse("<head><title>Player Notes</title></head><body>[notes_gethtml(ckey)]</body>","window=player_notes;size=700x400")
```
Comment (TorinoFermic) [inline code/modules/admin/player_notes.dm]: We are using to_chat(), they have to be replaced.

**Comment by GeneriedJenelle** [comment]
https://github.com/Baystation12/Baystation12/commit/c94b35dc00a0cb5d9e51ffc9378c6c79f6d753c1#diff-43cbbc3a371c8391b464134749f84924

Porting from here, I'll do it in a moment.

**Comment by GeneriedJenelle** [comment]
image

**edit:** So I've made the objs into datums to modernize the code a bit, and i've added a notes button to your PP menu. Only issue is the player panel for eris is custom and I didn't want to mess with it since you had custom code. If you guys get the modern version of the player panel you should be able to get the modern notes code,

That said, this version should be bug free and works similarly to modern code anyhow, just different layout. Serves it's function. Let me know if it's okay.

File: code/modules/admin/admin.dm [hunk 4]
```diff
@@ -29,12 +29,37 @@ proc/admin_notice(var/message, var/rights)
 /datum/admins/SDQL_update(var/const/var_name, var/new_value)
 	return 0
 
+/datum/admins/Topic(href, href_list)
+	..()
```
Comment (ACCount12) [inline code/modules/admin/admin.dm]: Move this to `topic.dm` so it's easy to clean it up later.

Comment (GeneriedJenelle) [reply code/modules/admin/admin.dm]: Now done!

**Comment by ACCount12** [comment]
Would have to be ported to datums later, but it's good enough for now.

File: code/modules/admin/topic.dm [hunk 5]
```diff
@@ -22,6 +22,26 @@
 		message_admins("[usr.key] has attempted to override the admin panel!")
 		return
 
+	//Player Notes
+	if(href_list["notes"])
```
Comment (nichlas0010) [inline code/modules/admin/topic.dm]: m a n e

**Pull request merged**